src/test/java/com/thealgorithms/datastructures/lists/SinglyLinkedListTest.java

```diff
@@ -0,0 +1,94 @@
+package com.thealgorithms.datastructures.lists;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class SinglyLinkedListTest {
+
+    /**
+     * Initialize a list with natural order values with pre-defined length
+     * @param length
+     * @return linked list with pre-defined number of nodes
+     */
+    private SinglyLinkedList createSampleList(int length) {
+        List<Node> nodeList = new ArrayList<>();
+        for (int i = 1; i <= length; i++) {
+            Node node = new Node(i);
+            nodeList.add(node);
+        }
+
+        for (int i = 0; i < length - 1; i++) {
+            nodeList.get(i).next = nodeList.get(i+1);
+        }
+
+        return new SinglyLinkedList(nodeList.get(0), length);
+    }
+
+    @Test
+    void detectLoop() {
+        //List has cycle
+        Node firstNode = new Node(1);
+        Node secondNode = new Node(2);
+        Node thirdNode = new Node(3);
+        Node fourthNode = new Node(4);
+
+        firstNode.next = secondNode;
+        secondNode.next = thirdNode;
+        thirdNode.next = fourthNode;
+        fourthNode.next = firstNode;
+
+        SinglyLinkedList listHasLoop = new SinglyLinkedList(firstNode, 4);
+        assertTrue(listHasLoop.detectLoop());
+
+        SinglyLinkedList listHasNoLoop = createSampleList(5);
+        assertFalse(listHasNoLoop.detectLoop());
+    }
+
+    @Test
+    void middle() {
```

Comment:
Please test middle on an empty list

Reply:
Added empty list check and list has only one node check